### PR TITLE
[2.9] Bump many dependencies to latest versions

### PR DESCRIPTION
**Pull request by JonCrowther**
Bump the following to their latest versions
- k3s
- etcdctl
- docker-machine-driver-linode
- docker-machine-driver-harvester
- helm
- kustomize

in Dockerfile.dapper I also updated
- k3d
- yq

### CODE REVIEW

File: build.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 webhookVersion: 104.0.0+up0.5.0-rc11
 provisioningCAPIVersion: 104.0.0+up0.3.0-rc.1
 cspAdapterMinVersion: 104.0.0+up4.0.0-rc9
-defaultShellVersion: rancher/shell:v0.2.1-rc.5
+defaultShellVersion: rancher/shell:v0.2.1-rc.7
```
Comment (mallardduck) [inline build.yaml:4]: 🙌  Thank you for adding this one.

**Review by macedogm — APPROVED** [review]

**Review by pjbgf — APPROVED** [review]

**Review by mallardduck — APPROVED** [review]

File: package/Dockerfile [hunk 2]
```diff
@@ -54,19 +54,19 @@ ENV CATTLE_PARTNER_CHART_DEFAULT_BRANCH=$PARTNER_CHART_DEFAULT_BRANCH
...
-ENV CATTLE_K3S_VERSION v1.30.1+k3s1
+ENV CATTLE_K3S_VERSION v1.30.2+k3s2
 ENV CATTLE_MACHINE_PROVISION_IMAGE rancher/machine:${CATTLE_MACHINE_VERSION}
-ENV CATTLE_ETCD_VERSION v3.5.12
+ENV CATTLE_ETCD_VERSION v3.5.14
 ENV LOGLEVEL_VERSION v0.1.6
 ENV TINI_VERSION v0.18.0
 ENV TELEMETRY_VERSION v0.6.2
-ENV DOCKER_MACHINE_LINODE_VERSION v0.1.11
+ENV DOCKER_MACHINE_LINODE_VERSION v0.1.12
```
Comment (snasovich) [inline package/Dockerfile:63]: @LBGarber , are you OK with bumping to `v0.1.12`? Looking at https://github.com/linode/docker-machine-driver-linode/releases/tag/v0.1.12 it's basically just dependency bumps but would like to get your 👍 on it.

File: package/Dockerfile [hunk 3]
```diff
@@ -54,19 +54,19 @@ ENV CATTLE_PARTNER_CHART_DEFAULT_BRANCH=$PARTNER_CHART_DEFAULT_BRANCH
...
 ENV LINODE_UI_DRIVER_VERSION v0.7.0
 # make sure the version number is consistent with the one at Line 100 of pkg/data/management/machinedriver_data.go
-ENV DOCKER_MACHINE_HARVESTER_VERSION v0.6.7
+ENV DOCKER_MACHINE_HARVESTER_VERSION v0.6.9
```
Comment (snasovich) [inline package/Dockerfile:66]: @bk201 / @innobead (cc: @yasker ), are you good with bumping Harvester machine driver version to `v0.6.9` for upcoming v2.9.0 release?

**Review by snasovich — COMMENTED** [review]
Redirected couple bumps to respective teams.

**Comment by ibrokethecloud** [comment]
any chance we could also include changes for packaging arm artifacts for harvester node driver? 

we started publishing multi-arch artifacts for harvester-node-driver with v0.6.8.

I have a draft PR here with changes needed for the same: https://github.com/rancher/rancher/pull/46089

Related issue: https://github.com/harvester/harvester/issues/6053

**Comment by JonCrowther** [comment]
@ibrokethecloud @snasovich I undid the bumps for docker-machine-driver-linode and docker-machine-driver-harvester in an effort to speed up the review time for this PR. They seem like they require more feedback and I want to get these dependencies merged for v2.9 before it releases

**Review by MbolotSuse — APPROVED** [review]

**Review by tomleb — APPROVED** [review]

**Review by mallardduck — APPROVED** [review]

File: package/Dockerfile (hunk 2 above)
Comment (lgarber-akamai) [reply package/Dockerfile]: Bumping docker-machine-driver-linode to `v0.1.12` should be fine in a future PR 👍

**Pull request merged**